plugins/modules/support_assist.py

```diff
@@ -0,0 +1,771 @@
+            if telemetry['telemetry_enabled'] is False and \
+                    settings_details['telemetry']['telemetry_enabled'] is not False:
+                telemetry_dict['telemetry_enabled'] = telemetry['telemetry_enabled']
+        if telemetry_dict != {}:
+            modify_dict["telemetry"] = telemetry_dict
+        return modify_dict
+
+    def is_support_assist_modify_required(self, settings_params, settings_details):
+        """
+        Check whether modification is required in support assist settings
+        """
+        modify_dict = {}
+        support_assist_keys = ["enable_download", "enable_remote_support",
+                               "automatic_case_creation", "connection_state"]
+        for key in support_assist_keys:
+            if settings_params.get(key) is not None and \
+                    settings_details[key] != settings_params[key]:
+                modify_dict[key] = settings_params[key]
+
+        if settings_params['enable_service'] is not None and \
+                settings_params['enable_service'] != settings_details['supportassist_enabled']:
+            modify_dict['enable_service'] = settings_params['enable_service']
+
+        return modify_dict
+
+    def validate_phone(self, phone):
+        if phone is not None and not phone.isnumeric():
+            error_msg = "The contact phone is invalid"
+            LOG.error(error_msg)
+            self.module.fail_json(msg=error_msg)
+
+    def is_support_assist_contact_modify_required(self, settings_params, settings_details, modify_dict):
+        """
+        Check whether modification is required in support assist settings
+        """
+        contact = {}
+        if settings_params['contact']:
+            if settings_params['contact']['primary']:
+                if settings_params['contact']['primary'] != settings_details['contact']['primary']:
+                    self.validate_phone(phone=settings_params['contact']['primary']['phone'])
+                    contact['primary'] = settings_params['contact']['primary']
+            if settings_params['contact']['secondary']:
+                if settings_params['contact']['secondary'] != settings_details['contact']['secondary']:
+                    self.validate_phone(phone=settings_params['contact']['secondary']['phone'])
+                    contact['secondary'] = settings_params['contact']['secondary']
+        if contact != {}:
+            modify_dict['contact'] = contact
+
+        return modify_dict
+
+    def get_support_assist_parameters(self):
+        return dict(
+            accepted_terms=dict(type='bool'),
+            automatic_case_creation=dict(type='bool'),
+            connection=dict(
+                type='dict', options=dict(
+                    gateway_endpoints=dict(
+                        type='list', elements='dict', options=dict(
+                            enabled=dict(type='bool', default=True),
+                            gateway_host=dict(type='str'),
```

Comment:
can we make gateway_host in gateway_endpoints as a required param

Reply:
done.